integration_tests/test_employment.py

```diff
@@ -73,3 +75,153 @@ def test_movers_change_employment(populations):
+            tracked_live_populations, time_step, "employer_income_propensity"
+        )
+
+        # Find those that change employers
+        static_jobs_idx = employer_ids_before.index[employer_ids_before == employer_ids_after]
+
+        assert (
+            employer_propensity_before.loc[static_jobs_idx]
+            == employer_propensity_after.loc[static_jobs_idx]
+        ).all()
+
+
+def test_personal_income_propensity_is_constant(tracked_live_populations):
+    starting_personal_income_propensity = tracked_live_populations[min(TIME_STEPS_TO_TEST)][
+        "personal_income_propensity"
+    ]
+    ending_personal_income_propensity = tracked_live_populations[max(TIME_STEPS_TO_TEST)][
+        "personal_income_propensity"
+    ]
+    # Get index for sims that exist in both state tables
+    present_sims_idx = starting_personal_income_propensity.index.intersection(
+        ending_personal_income_propensity.index
+    )
+
+    assert (
+        starting_personal_income_propensity.loc[present_sims_idx]
+        == ending_personal_income_propensity.loc[present_sims_idx]
+    ).all()
+
+
+def test_income_updates(tracked_live_populations):
+    for time_step in range(max(TIME_STEPS_TO_TEST)):
+        (
+            employer_ids_before,
+            income_before,
+            employer_ids_after,
+            income_after,
+        ) = get_consecutive_time_step_employer_ids_and_test_column_for_comparable_sims(
+            tracked_live_populations, time_step, "income"
+        )
+
+        # Find those that change employers
+        changed_jobs_idx = employer_ids_before.index[
+            employer_ids_before != employer_ids_after
+        ]
+
+        assert (
+            income_before.loc[changed_jobs_idx] != income_after.loc[changed_jobs_idx]
+        ).all()
+
+
+def test_no_income_uupdate(tracked_live_populations):
+    for time_step in range(max(TIME_STEPS_TO_TEST)):
+        (
+            employer_ids_before,
+            income_before,
+            employer_ids_after,
+            income_after,
+        ) = get_consecutive_time_step_employer_ids_and_test_column_for_comparable_sims(
+            tracked_live_populations, time_step, "income"
+        )
+
+        # Find those that change employers
+        static_jobs_idx = employer_ids_before.index[employer_ids_before == employer_ids_after]
+
+        assert (income_before.loc[static_jobs_idx] == income_after.loc[static_jobs_idx]).all()
+
+
+def test_income_updates_when_age_changes(tracked_live_populations):
+    # This test checks whether simulants who have a birthdays and therefore will have a new income distribution
+    # Note: The other tests for income implementation are contigent on comparing simulants who are the same age and
+    #  therefore ignore this edge case.
+    for time_step in range(max(TIME_STEPS_TO_TEST)):
+        pop_before = tracked_live_populations[time_step]
+        pop_after = tracked_live_populations[time_step]
```

Comment:
Isn't this comparing a population with itself?

Add an assert that there are `birthday_sims`.